src/commands/azureServiceOperators/installAzureServiceOperator.ts

```diff
@@ -0,0 +1,158 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from 'vscode-azureextensionui';
+import AksClusterTreeItem from '../../tree/aksClusterTreeItem';
+import { getExtensionPath, longRunning } from '../utils/host';
+import {
+    getWebviewContent,
+    getOperatorsPod,
+    installCertManager,
+    installIssuerCert,
+    getAzureServicePrincipal,
+    installOperatorSettings,
+    checkCertManagerRolloutStatus,
+    installOlmCrd,
+    installOlm,
+    installOperator
+} from './helpers/azureservicehelper';
+import * as clusters from '../utils/clusters';
+import { InstallationResponse } from './models/installationResponse';
+
+export default async function installAzureServiceOperator(
+    context: IActionContext,
+    target: any
+): Promise<void> {
+    const kubectl = await k8s.extension.kubectl.v1;
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+    const clusterExplorer = await k8s.extension.clusterExplorer.v1;
+
+    if (cloudExplorer.available && kubectl.available) {
+        const clusterTarget = cloudExplorer.api.resolveCommandTarget(target);
+
+        if (clusterTarget && clusterTarget.cloudName === "Azure" &&
+            clusterTarget.nodeType === "resource" && clusterTarget.cloudResource.nodeType === "cluster" &&
+            clusterExplorer.available) {
+
+            const aksCluster = clusterTarget.cloudResource as AksClusterTreeItem;
+            await install(aksCluster);
+            clusterExplorer.api.refresh();
+        } else {
+            vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+        }
+    }
+}
+
+async function install(
+    aksCluster: AksClusterTreeItem
+): Promise<void> {
+
+    const kubectl = await k8s.extension.kubectl.v1;
```

Comment:
You already get this in the calling function (though maybe you don't use it) - why not check it there so that everything is checked up top and the resolved, known-good API objects are passed down to here?

```
function installAzureServiceOperator(...) {
    const kubectlAPI = await k8s.extension.kubectl.v1;
    const cloudExplorerAPI = await k8s.extension.cloudExplorer.v1;
    const clusterExplorerAPI = await k8s.extension.clusterExplorer.v1;

    // check prerequisites
    if (!kubectlAPI.available) {
        // error and exit
    }
    if (!clusterExplorerAPI.available) {
        // error and exit
    }
    if (!cloudExplorerAPI.available) {
        // error and exit
    }

    const kubectl = kubectlAPI.api;
    const clusterExplorer = clusterExplorerAPI.api;
    const cloudExplorer = cloudExplorerAPI.api;

    // Now use kubectl, clusterExplorer and cloudExplorer freely

    install(aksCluster, kubectl);
}

function install(
  aksCluster: AksClusterTreeItem,
  kubectl: k8s.KubectlV1
) { ... }
```